backend/src/core/session/session.middleware.ts

```diff
@@ -23,24 +23,56 @@ export class SessionMiddleware implements NestMiddleware {
   }
 
   async use(
-    req: FastifyRequest["raw"],
-    res: FastifyReply["raw"],
+    req: FastifyRequest,
+    res: FastifyReply,
     next: () => void,
   ) {
     try {
+      // Extract query parameters from the URL
+      const queryParams: Record<string, string> = {};
```

Comment:
Avec les modifs que j'ai ajoutées sur le dernier commit il me semble qu'on peut simplifier cette partie du code et la rendre plus idiomatique

```suggestion
    // Extract query parameters directly from Fastify's parsed query
    const queryParams = req.query as Record<string, string>;
    
    // Convert request headers to a simple object format
    const headers: Record<string, string> = {};
    Object.keys(req.headers).forEach((key) => {
      // Handle potential array headers by taking the first value
      const headerValue = req.headers[key];
      headers[key] = Array.isArray(headerValue)
        ? headerValue[0]
        : (headerValue as string);
    });

    // Get the path directly from Fastify
    const pathPrefix = req.url.split("/")[0] || "";

    // Call the session service with all authentication data
    const session = await this.entServiceClient.sessionFind({
      cookies: req.headers.cookie,
      headers,
      params: queryParams,
      pathPrefix,
      path: req.url,
    });
```